core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -223,6 +223,7 @@ object Battle {
             return
         var amountToAdd = amount
         if(thisCombatant.getCivInfo().policies.isAdopted("Military Tradition")) amountToAdd = (amountToAdd * 1.5f).toInt()
+        if (thisCombatant.unit.hasUnique("50% Bonus XP gain")) amountToAdd = (amountToAdd * 1.5f).toInt()
```

Comment:
Should probably be it's own function, units. Promotion.addXp(amount)

Reply:
Oh, I think you misunderstood me...
I meant that the Military Tradition and the 50% Bonus Xp Gain should go in the 'Add XP' function, since now they're checked in 2 separate places in the code :)